script/tests/test_formatting_locks.sh

```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+set -euo pipefail
+source ./script/functions.sh
+REPO_DIRS=$(find ./formatting_locks -type d -mindepth 2)
+for REPO_DIR in $REPO_DIRS
+do
+    HEAD=$(cat "$REPO_DIR/HEAD")
+    FILES=$(cat "$REPO_DIR/files")
+    REPO=$(echo "$REPO_DIR" | sed 's/.\/formatting_locks\///')
+    (
+    cd "$(mktemp -d)"
```

Comment:
Could this use a static directory instead of a new temp dir each time? If this is checking out the same versions each time it seems wasteful to repeat that for each run of this script when that is easily cacheable.

Reply:
> Could this use a static directory instead of a new temp dir each time? If this is checking out the same versions each time it seems wasteful to repeat that for each run of this script when that is easily cacheable.

Yeah I think that's reasonable